### ISSUE TITLE
We should consider requiring a feature flag for emulated f16 usage

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#7970]
We currently emit a warning if you try to use float16/bfloat16 without native support, but (surprise!) people ignore warnings (or they are swallowed by the build system), and then wonder why the code is so slow.

I'd propose that we instead consider removing the warnings, and add a new Feature flags named (say) for "slow_emulate_f16" and "slow_emulate_bf16":
- using f16 or bf16 without a proper enabling flag of some sort is an error,  not a warning
- specifying these flags in combination with a native f16 flag (eg arm_fp16, f16c) is an error

This requires explicit user understanding that they are building code that is functional, but not performant.

**Comment by steven-johnson** [comment]
(Additionally, it appears that when targeting Windows and using emulated, the relevant glue code for f16<->f32 conversions is referenced by LLVM but not emitted by the Linker, leading to confusing link errors in some build configs. This appears to be a Clang error so it's not clear if we should try to address it, but it might be good to document it while we are there. There are ways we could work around it but it's probably better to just tell people to specify `f16c` for these targets, since most mainstream x86-64 chips for the past ~10 years support it.)

**Comment by abadams** [comment]
We may need to distinguish between emulating f16 conversions (which we don't need to do on most x86 processors) and emulating f16 math using f32 math (which we *do* need to do on all but the newest x86 processors).

**Comment by steven-johnson** [comment]
Yeah, I didn't think about that part. Full f16 requires AVX10 or similar, right?

**Comment by abadams** [comment]
I think Sapphire Rapids has it, but I think it'll be avx10 before it's widely available.

**Comment by abadams** [comment]
Note that just making the performance warning louder is not a good solution, because you don't want warning spam when you're intentionally compiling a pipeline that emulates float16.
